src/base/utils/fs.cpp

```diff
@@ -128,17 +132,26 @@ bool Utils::Fs::smartRemoveEmptyFolderTree(const QString &path)
               , [](const QString &l, const QString &r) { return l.count('/') > r.count('/'); });
 
     for (const QString &p : asConst(dirList)) {
-        // remove unwanted files
-        for (const QString &f : deleteFilesList) {
-            forceRemove(p + f);
-        }
-
-        // remove temp files on linux (file ends with '~'), e.g. `filename~`
         const QDir dir(p);
+        if (!dir.isEmpty(QDir::AllDirs | QDir::NoDotAndDotDot))
+            continue;
```

Comment:
I would add empty line below, please.

Reply:
Sorry, I don't understand this one. Don't you remove folder if it contains subfolders?

Reply:
Yes. The search is done from deepest folder outwards. But if a deeper folder wasn't removed, then this one shouldn't be removed (or have useless files removed).
```const QStringList tmpFileList = dir.entryList(QDir::Files);```
holds only filenames.

Reply:
It would be nice to have such a comment at this point to avoid misunderstandings during future reviews.

Reply:
Done.